src/vivarium/framework/population/manager.py

```diff
@@ -399,19 +394,127 @@ def _create_simulants(
+            raise PopulationError(
+                f"Requested attribute(s) {non_existent_attributes} not in population table. "
+                "This is likely due to a failure to require some columns, randomness "
+                "streams, or pipelines when registering a simulant initializer, a value "
+                "producer, or a value modifier. NOTE: It is possible for a run to "
+                "succeed even if resource requirements were not properly specified in "
+                "the simulant initializers or pipeline creation/modification calls. This "
+                "success depends on component initialization order which may change in "
+                "different run settings."
+            )
+
+        attributes_list: list[pd.Series[Any] | pd.DataFrame] = []
+
+        # batch simple attributes and pop right off the backing data
+        simple_attributes = [
+            name
+            for name, pipeline in self._attribute_pipelines.items()
+            if name in attributes_to_include and pipeline.is_simple
+        ]
+        if simple_attributes:
+            attributes_list.append(self._population.loc[idx, simple_attributes])
+
+        # handle remaining non-simple attributes one by one
+        remaining_attributes = [
+            attribute
+            for attribute in attributes_to_include
+            if attribute not in simple_attributes
+        ]
+        for name in remaining_attributes:
+            pipeline = self._attribute_pipelines[name]
+            values = self._population.loc[idx, name] if pipeline.is_simple else pipeline(idx)
+            if isinstance(values, pd.Series):
+                if values.name != name:
+                    self.logger.warning(
+                        f"The '{name} attribute pipeline returned a pd.Series with a "
+                        f"different name '{values.name}'. For the column being added to the "
+                        "population state table, we will use '{name}'."
+                    )
+                values.name = name
+            attributes_list.append(values)
+
+        df = (
+            pd.concat(attributes_list, axis=1) if attributes_list else pd.DataFrame(index=idx)
+        )
+
+        duplicate_columns = df.columns[df.columns.duplicated()].tolist()
+        if duplicate_columns:
+            raise PopulationError(
+                f"Population table has duplicate column names: {duplicate_columns}. "
+                "This is likely due to an AttributePipeline producing a pd.Dataframe with the "
+                "same column name(s) that some Component has in its `columns_created` property."
+            )
+
+        # TODO: We need to handle column ordering as well as potential column name
+        # collisions. Consider multi-indexes as a way to handle it both, but only
+        # if the ux doesn't change. Else, we can prepend the pipeline name to the
+        # column names that it produces, e.g. "acceleration.x", "acceleration.y", etc.
```

Comment:
@rmudambi I added the query into this method as discussed, but I honestly don't really understand what we gained by doing that? I _think_ the idea was that we would gain the ability to parse the query string to know which pipelines to add to the state table, but then agreed that we won't do that unless necessary later. So should I keep it here or just move it back?